pkg/apis/extensions/types.go

```diff
@@ -332,6 +332,9 @@ type RollingUpdateDeployment struct {
 }
 
 type DeploymentStatus struct {
+	// The generation observed by the deployment controller.
+	ObservedGeneration int `json:"observedGeneration,omitempty"`
```

Comment:
> All internal integer types are int ...

If it's an increasing number, it should be `int64`, see ReplicaSetStatus.ObservedGeneration or HorizontalPodAutoscalerStatus.ObservedGeneration in pkg/apis/extensions/types.go